### DISCUSSION TITLE
Como começar projetos grandes em NodeJs?

### DISCUSSION THREAD

**Discussion by jeffersonalionco** [frontendbr/forum#2434] [Discussão]
Olá pessoal meu nome é **_[Jefferson L. Alionco](https://github.com/jeffersonalionco)_** , creio que não é um divida só minha mais de muitas pessoas que estão iniciando na aréa. O que preciso para inicializar um grande projeto? Se preciso de uma equipe ou não? Que tipo de projeto seria bom eu aprender na carreira de NodeJs? Quais Frameworks é mais recomendavel aprender e usar?

É coisas basicas para profissionais, mais muitos iniciantes não conseguem prosseguir nos projetos por falta de conhecimento.

**Comment by igorlvicente** [comment]
Você está partindo do lugar errado.
Você não parte de NodeJS e cria um projeto grande. Você acha um problema e tenta resolvê-lo.
Se o projeto vai ser grande ou pequeno, depende do tamanho do problema.
Se vai usar NodeJS ou não, depende de vários fatores. Um desses muitos fatores pode ser o "tecnologia que o time está acostumado".

>Que tipo de projeto seria bom eu aprender na carreira de NodeJs?
>Quais Frameworks é mais recomendavel aprender e usar?

Se você quer aprender pra arrumar emprego, o que eu faria se tivesse que começar de novo seria:
1. Procure vagas para desenvolvedor NodeJS
2. Selecione as vagas em que a empresa trabalha num produto legal (ou pelo menos exclua as que fazem um produto chato). Eu selecionaria umas 30 vagas ou mais.
3. Veja o que é mais comum entre elas nos requisitos. Provavelmente vai ter um framework e um conjunto de libs/ferramentas que vai se destacar. Escolha ele.
4. Escolha, dessas 30+ empresas, a que o produto pareça ser mais útil pra você
5. Tente fazer você um sistema que resolva o mesmo problema

Ou seja, no passo 3 você vai descobrir a tecnologia e no passo 4 você vai descobrir no que trabalhar. Depois que se tem um objetivo/plano, é só seguir.
A partir desse objetivo você vai criando perguntas mais específicas e, consequentemente, mais fáceis e diretas de serem respondidas.
Você deixará de ter perguntas do tipo "Qual tecnologia é melhor?" pra ter perguntas do tipo "como eu implemento essa funcionalidade usando essa tecnologia que eu escolhi?".
